src/classes/AFFL_AccRecordType_TDTM.cls

```diff
@@ -156,13 +157,15 @@ public class AFFL_AccRecordType_TDTM extends TDTM_Runnable {
                 }
             }
             //A.2 - Change all affiliations that are not primary any more.
-            if(contactIDs.size() > 0 && accountIDs.size() > 0) {
+            if( contactIDs.size() > 0 && accountIDs.size() > 0) {
                 List<Affiliation__c> afflsMadePrimary = [select Contact__c, Account__c, Affiliation_Type__c
                           from Affiliation__c where Contact__c IN :contactIDs and Account__c IN :accountIDs];
                 AFFL_MultiRecordType_TDTM afflMulti = new AFFL_MultiRecordType_TDTM();
                 afflMulti.uncheckOtherPrimariesSameType(afflsMadePrimary, dmlWrapper);
             }
         }
+
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.CON_PrimaryAffls_TDTM_After_Update, true);
```

Comment:
@skosgi1 @lifewithryan I'd also like to see either a new unit test that tests that duplicate primary affiliation records aren't created when an account record type is updated (or perhaps an existing test). The fact that our unit tests didn't catch this bug to begin with is a pretty good indication to me that we should have one.

Reply:
@robertbwatson On it!

Reply:
@robertbwatson @lifewithryan Added a new test method for duplicate check.